suprb/optimizer/rule/generation_operator/base.py

```diff
@@ -0,0 +1,25 @@
+from abc import ABCMeta, abstractmethod
+from suprb.rule.matching import MatchingFunction, OrderedBound
+from suprb.rule import Rule
+from suprb.base import BaseComponent
+from suprb.utils import RandomState
+
+
+class GenerationOperator(BaseComponent, metaclass=ABCMeta):
+    def __init__(self, matching_type: MatchingFunction = None):
+        self.matching_type = matching_type
+
+    @property
+    def matching_type(self):
+        return self._matching_type
+
+    @matching_type.setter
+    def matching_type(self, matching_type):
+        self._matching_type = matching_type
+        if isinstance(self.matching_type, OrderedBound):
+            self.mutate_bounds = self.ordered_bound
+            self._crossover = self.ordered_bound
```

Comment:
we really shouldn't do that. this would give an object of a crossover class the method mutate_bounds that then does not mutate bounds

Reply:
you could call the function `generate` to make it more generic. Maybe, a more specific name would be more fitting? I am unsure

Reply:
Called it `execute`